client/components/app.jsx

```diff
@@ -24,10 +25,20 @@ class App extends React.Component {
       .catch(error => console.error('Fetch error', error));
   }
 
+  getAverageGrade(grades) {
+    let average = null;
+    let averageTotal = null;
+    for (let i = 0; i < grades.length; i++) {
+      average += grades[i].grade;
+    }
+    averageTotal = Math.ceil(average / grades.length);
+    return averageTotal;
```

Comment:
So solve the `NaN` warning you need a conditional to check the length of the grades array before calculating the average. The length of the array may be `0` and `0 / 0` is `NaN`. You can return `null` for example.

Reply:
Okay, I'll implement the conditional.